### PR TITLE
[feature](nereids) let set operation syntax campatible with lagecy planner

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Though this syntax doesn't get suppoted in many other systems since the order by clause here almost redandunt and useless but we have to keep consistent with the legacy doris syntax

Here is a example:

```sql
SELECT * FROM (select k1, k3 from setOperationTableNotNullable order by k3 union all 
            select k1, k5 from setOperationTable) t;
```

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.76 seconds
 load time: 649 seconds
 storage size: 17122998827 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230105194418_clickbench_pr_74497.html

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -53,11 +57,12 @@ planType
...
 //  -----------------Query-----------------
 query
-    : cte? queryTerm queryOrganization
+    : {!doris_legacy_SQL_syntax}? cte? queryTerm queryOrganization
+    | {doris_legacy_SQL_syntax}? queryTerm
     ;
 
 queryTerm
-    : queryPrimary                                                                       #queryTermDefault
+    : queryPrimary                                                                      #queryTermDefault
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:65]: add back removed blank

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 2]
```diff
@@ -70,15 +75,17 @@ setQuantifier
 queryPrimary
     : querySpecification                                                    #queryPrimaryDefault
     | TABLE multipartIdentifier                                             #table
-    | LEFT_PAREN query RIGHT_PAREN                                          #subquery
+    | LEFT_PAREN query RIGHT_PAREN                                      #subquery
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:78]: ditto

File: regression-test/suites/nereids_syntax_p0/set_operation.groovy [hunk 3]
```diff
@@ -242,4 +241,9 @@ suite("test_nereids_set_operation") {
     INTERSECT
     SELECT k1 FROM setOperationTable WHERE k2 > 0)
     """
+
+    order_qt_select43 """
+        SELECT * FROM (select k1, k3 from setOperationTableNotNullable order by k3 union all 
+            select k1, k5 from setOperationTable) t;
+    """
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/set_operation.groovy:248]: add more case, to test all situation of legacy planner's syntax. such as set operation, w/ and w/o parenthesis, w/ and w/o cte and w/ and w/o query organization

Comment (Kikyou1997) [reply regression-test/suites/nereids_syntax_p0/set_operation.groovy:248]: done

**Review by morrySnow — APPROVED** [review]

**Pull request merged**